resources/assets/lib/chat/join-channel.tsx

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import BigButton from 'components/big-button';
+import { Spinner } from 'components/spinner';
+import UserCardBrick from 'components/user-card-brick';
+import UserJson from 'interfaces/user-json';
+import { action, computed, makeObservable } from 'mobx';
+import { observer } from 'mobx-react';
+import { isInputKey } from 'models/chat/create-announcement';
+import core from 'osu-core-singleton';
+import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
+
+type Props = Record<string, never>;
+
+interface InputContainerProps {
+  error: boolean;
+  labelKey?: string;
+}
+
+const BusySpinner = ({ busy }: { busy: boolean }) => (
+  <div className='chat-join-channel__spinner'>
+    {busy && <Spinner />}
+  </div>
+);
+
+// TODO: look at combining with ValidatingInput
+const InputContainer = observer((props: React.PropsWithChildren<InputContainerProps>) => (
+  <div className={classWithModifiers('chat-join-channel__input-container', { error: props.error })}>
+    {props.labelKey && (
+      <label className='chat-join-channel__input-label'>{osu.trans(props.labelKey)}</label>
```

Comment:
the `div` and `label` should be swapped so clicking anywhere on the area will focus on the input. Or at least give the label appropriate name property